### PR TITLE
Add Filter by Number of Citations

### PR DESCRIPTION
**Pull request by ratuspro**

### CODE REVIEW

File: tests/unit/filters/test_filter_by_citations.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+paper_with_citations = Paper("id_with_citations")
+for i in range(0,PAPER_CITATIONS):
+    paper_with_citations.citations_ids.add(f"cited_paper_{i}")
+
+paper_without_citations = Paper("id_without_citations")
+
+class TestCitationsFilter(unittest.TestCase):
+    def test_citations_filter_incorrect_min_and_max(self):
+        filter = FilterByCitations(PAPER_CITATIONS + 1, PAPER_CITATIONS - 1)
+        self.assertRaises(Exception)
```
Comment (RicardoEPRodrigues) [inline tests/unit/filters/test_filter_by_citations.py]: This test is not doing anything. I propose the following change:

```python
def test_citations_filter_incorrect_min_and_max(self):
    filter = FilterByCitations(PAPER_CITATIONS + 1, PAPER_CITATIONS - 1)
    with self.assertRaises(Exception):
        filter.is_valid(paper_with_citations)
```

Using this solution also shows that the code isn't raising an exception. So changes are needed in the `is_valid()` function.

**Review by RicardoEPRodrigues — CHANGES_REQUESTED** [review]
An exception needs to be launched when the min is greater than the max, and the test evaluating that needs to be updated.

**Comment by ratuspro** [comment]
@RicardoEPRodrigues thanks for solving the exception issue. I will apply it to the other tests.

If this PR is ok, can you please accept it?

**Review by RicardoEPRodrigues — APPROVED** [review]
Now it seems OK.

**Pull request merged**